aridanalysis/aridanalysis.py

```diff
@@ -282,40 +287,116 @@ def arid_logreg(data_frame, response, features=[], type="binomial", model="addit
+      
       
 
     Returns
     -------
-    pandas.DataFrame
-      Data frame with 4 columns: 'features', 'p-value', 'significant', 'interpretation'
-    String
-      Which family was used in the generalized linear regression model based on an overdispersion and fitting analysis
-      
+    sklearn.linear_model
+        A fitted sklearn model configured with the chosen input parameters
+    statsmodels.regression.linear_model
+        A fitted statsmodel configured with the chosen input parameters
+
 
     Examples
     --------
     >>> from aridanalysis import aridanalysis
     >>> aridanalysis.arid_countreg(df, income, features = [feat1, feat5] ,"additive")
     """
-    return None
+    assert isinstance(con_features, list), "ERROR: INVALID LIST INTPUT PASSED"
+    assert isinstance(cat_features, list), "ERROR: INVALID LIST INTPUT PASSED"
+    
+    
+    #Deal with the features column
+    if len(con_features) == 0:
+        con_features = data_frame.drop(columns=[response]).select_dtypes('number').columns.tolist()
+    if len(cat_features) == 0:
+        cat_features = data_frame.drop(columns=[response]).select_dtypes(['category', 'object']).columns.tolist()
+    
+    
+    assert isinstance(data_frame, pd.DataFrame), errors.INVALID_DATAFRAME
+    assert not data_frame.empty , errors.EMPTY_DATAFRAME
+    assert response in data_frame.columns.tolist(), errors.RESPONSE_NOT_FOUND
+    assert all(item in data_frame.columns.tolist() for item in con_features), "ERROR: CONTINUOUS VARIABLE(S) NOT IN DATAFRAME"
+    assert all(item in data_frame.columns.tolist() for item in cat_features), "ERROR: CATEGORICAL VARIABLE(S) NOT IN DATAFRAME"
+    assert ptypes.is_integer_dtype(data_frame[response].dtype), "ERROR: INVALID RESPONSE DATATYPE FOR COUNT REGRESSION: MUST BE TYPE INT"
+    assert model in ["additive", "interactive"], "ERROR: INVALID MODEL PASSED"
+    assert ptypes.is_numeric_dtype(type(alpha)), errors.INVALID_ALPHA_INPUT
+    
+    
+  
+    #Scikit Learn Model 
+    if len(cat_features) != 0:
+        X_sk = data_frame[con_features + cat_features]
+        y_sk = data_frame[response]
+        preprocessor = make_column_transformer((OneHotEncoder(handle_unknown="ignore"), cat_features))
+        pipeline = make_pipeline(preprocessor, PoissonRegressor(alpha=alpha, fit_intercept=True,))
+        sk_model = pipeline.fit(X_sk,y_sk)
+    else:
+        X_sk = data_frame[con_features]
+        y_sk = data_frame[response]
+        pipeline = make_pipeline(PoissonRegressor(alpha=0, fit_intercept=True, max_iter=100))
+        sk_model = pipeline.fit(X_sk,y_sk)    
+    #Aditive inferential model 
+    if model == "additive":
+        cat_features =["C(" + i + ")" for i in cat_features]
+        con_list = "".join([f"{i}" if i is con_features[0] else f" + {i}"for i in con_features])
+        cat_list = "".join([f"{i}" if i is cat_features[0] else f" + {i}"for i in cat_features])
+        if len(cat_list) > 0:
```

Comment:
Maybe a comment here in the future, this is only because of the weird statsmodel interface allowing R style regression function syntax.